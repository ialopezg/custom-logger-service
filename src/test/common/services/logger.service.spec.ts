const { LoggerService } = require("../../../common");

describe("LoggerService", () => {
  describe("Default configuration", () => {
    let logger: any;

    const config = {
      useAppName: true,
      useContext: true,
      useFormat: true,
      useEvent: true,
      usePID: true,
      useTimestamp: true,
      usePadding: true,
    };

    beforeAll(() => {
      logger = new LoggerService();
    });

    it("should  setup a logger instance with default options", () => {
      expect(logger["_options"]).toStrictEqual(config);
      expect(config).toMatchSnapshot();
    });
  });

  describe("Custom configuration", () => {
    let logger: any;

    const config = Object.assign({
      useAppName: "TestApp",
    });

    beforeAll(() => {
      logger = new LoggerService(config);
    });

    it("should setup a logger instance with custom options", () => {
      expect(logger["_options"]).toStrictEqual({
        ...config,
        useContext: true,
        useFormat: true,
        useEvent: true,
        usePID: true,
        useTimestamp: true,
        usePadding: true,
      });
      expect(config).toMatchSnapshot();
    });
  });

  describe("Event log messages", () => {
    let logger: any;

    const config = Object.assign({
      useAppName: "TestApp",
      useContext: "TestLoader",
    });

    beforeAll(() => {
      logger = new LoggerService(config);
    });

    it("should print a debug message with gray color", () => {
      const spy = jest.spyOn(logger, "debug").mockImplementation(() => {});

      const message = "Console Debug testing message";
      logger.debug(message);

      expect(logger.debug).toBeCalled();
      expect(spy.mock.calls[0][0]).toContain(message);
      expect(true).toMatchSnapshot();

      spy.mockRestore();
    });

    it("should print an error message with red color", () => {
      const spy = jest.spyOn(logger, "error").mockImplementation(() => {});

      const message = "Console Error testing message";
      logger.error(message);

      expect(logger.error).toBeCalled();
      expect(spy.mock.calls[0][0]).toContain(message);
      expect(spy.mock.calls[0][0]).toMatchSnapshot();

      spy.mockRestore();
    });

    it("should print an info message with green color", () => {
      const spy = jest.spyOn(logger, "info").mockImplementation(() => {});

      const message = "Console Info testing message";
      logger.info(message);

      expect(logger.info).toBeCalled();
      expect(spy.mock.calls[0][0]).toContain(message);
      expect(spy.mock.calls[0][0]).toMatchSnapshot();

      spy.mockRestore();
    });

    it("should print a warning message with yellow color", () => {
      const spy = jest.spyOn(logger, "warn").mockImplementation(() => {});

      const message = "Console Warn testing message";
      logger.warn(message);

      expect(logger.warn).toBeCalled();
      expect(spy.mock.calls[0][0]).toContain(message);
      expect(spy.mock.calls[0][0]).toMatchSnapshot();

      spy.mockRestore();
    });

    it("should print a log message with grey color", () => {
      const spy = jest.spyOn(logger, "log").mockImplementation(() => {});

      const message = "Console Log testing message";
      logger.log(message);

      expect(logger.log).toBeCalled();
      expect(spy.mock.calls[0][0]).toContain(message);
      expect(true).toMatchSnapshot();

      spy.mockRestore();
    });
  });
});
