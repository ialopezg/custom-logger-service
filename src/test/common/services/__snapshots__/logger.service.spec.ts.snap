// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggerService Configuration should print a debug message with gray color 1`] = `true`;

exports[`LoggerService Configuration should print a log message with grey color 1`] = `true`;

exports[`LoggerService Configuration should print a warning message with yellow color 1`] = `"Console Warn testing message"`;

exports[`LoggerService Configuration should print an error message with red color 1`] = `"Console Error testing message"`;

exports[`LoggerService Configuration should print an info message with green color 1`] = `"Console Info testing message"`;

exports[`LoggerService Configuration should setup a logger instance with custom options 1`] = `
Object {
  "useAppName": "TestApp",
}
`;

exports[`LoggerService Configuration should setup configurations for a logger instance 1`] = `
Object {
  "useAppName": "TestApp",
}
`;

exports[`LoggerService Custom configuration event log messages should print a debug message with gray color 1`] = `true`;

exports[`LoggerService Custom configuration event log messages should print a log message with grey color 1`] = `true`;

exports[`LoggerService Custom configuration event log messages should print a warning message with yellow color 1`] = `"Console Warn testing message"`;

exports[`LoggerService Custom configuration event log messages should print an error message with red color 1`] = `"Console Error testing message"`;

exports[`LoggerService Custom configuration event log messages should print an info message with green color 1`] = `"Console Info testing message"`;

exports[`LoggerService Custom configuration should setup a logger instance with custom options 1`] = `
Object {
  "useAppName": "TestApp",
}
`;

exports[`LoggerService Default configuration should  setup a logger instance with default options 1`] = `
Object {
  "useAppName": true,
  "useContext": true,
  "useEvent": true,
  "useFormat": true,
  "usePID": true,
  "usePadding": true,
  "useTimestamp": true,
}
`;

exports[`LoggerService default configuration should show a log event with default options 1`] = `"Console Debug testing message"`;

exports[`LoggerService event log messages should print a debug message with gray color 1`] = `true`;

exports[`LoggerService event log messages should print a log message with grey color 1`] = `true`;

exports[`LoggerService event log messages should print a warning message with yellow color 1`] = `"Console Warn testing message"`;

exports[`LoggerService event log messages should print an error message with red color 1`] = `"Console Error testing message"`;

exports[`LoggerService event log messages should print an info message with green color 1`] = `"Console Info testing message"`;
